src/autoformatengine.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import Range from '../engine/model/range.js';
+import TextProxy from '../engine/model/textproxy.js';
+
+export default class AutoformatEngine {
+	/**
+	 * Creates listener triggered on `change` event in document. Calls callback when inserted text matches regular expression.
+	 *
+	 * @param {core.editor.Editor} editor Editor instance.
+	 * @param {Regex} pattern Regular expression to exec on just inserted text.
+	 * @param {Function|String} callbackOrCommand Callback to execute or command to run when text is matched.
+	 * In case of providing callback it receives following parameters:
+	 * * {engine.model.Batch} batch Newly created batch for autoformat changes.
+	 * * {Object} match RegExp.exec() result of matching pattern to inserted text.
+	 */
```

Comment:
Example of usage (creating own autoformat) added to that documentation would be nice.